### ISSUE TITLE
Implement test case for IOUtils.java (and ideally everything else)

### ISSUE THREAD

**Issue by haowu4** [IllinoisCogComp/illinois-cogcomp-nlp#224]
Many resources related functions are obscure and hard to reason with. (Jar file handling, Windows issues). For example, without an unit-test, I almost made a mistake by connecting JarEntry with Windows File.SEPARATOR.

Without unit-test, no one can be 100% sure if we are fixing bugs or introducing more.

We should at least have unittest for those tricky functions and critical ones. And I suggest we add a [Corveralls](https://coveralls.io) service to monitor the progress. Also, otherwise we are not taking the full advantage of CI.

**Comment by haowu4** [comment]
Also, does Semaphore provide Windows env testing?

**Comment by danyaljj** [comment]
>  does Semaphore provide Windows env testing?
it does not. But @bhargav has installed  AppVoyer which has Win env.

**Comment by bhargav** [comment]
Yes. I have setup Windows CI to only build changes to the master branch. Currently the unit tests fail. It might be good to (spend some time to) debug in some Windows machine once and make sure we do not regress.

https://ci.appveyor.com/project/bhargavm/illinois-cogcomp-nlp

**Comment by danyaljj** [comment]
@haowu4 are you planning to address this issue yourself?

**Comment by mssammon** [comment]
@haowu4 let me know if you want to break this up into multiple smaller issues
